src/contract.ts

```diff
@@ -0,0 +1,279 @@
+import { ABI, ABIParameterInfo } from './abi';
+
+export interface FunctionCallOptions {
+    /** max amount of gas that method call can use */
+    gas?: BN;
+    /** amount of NEAR (in yoctoNEAR) to send together with the call */
+    attachedDeposit?: BN;
+    /**
+     * Metadata to send the NEAR Wallet if using it to sign transactions.
+     * @see {@link RequestSignTransactionsOptions}
+     */
+    walletMeta?: string;
+    /**
+     * Callback url to send the NEAR Wallet if using it to sign transactions.
+     * @see {@link RequestSignTransactionsOptions}
+     */
+    walletCallbackUrl?: string;
+}
+
+function deserializeJSON(response: Uint8Array): any {
+    return JSON.parse(Buffer.from(response).toString());
+}
+
+function serializeJSON(input: any): Buffer {
+    return Buffer.from(JSON.stringify(input));
+}
+
+// Helper function so that
+function ignoreSerialization(input: Buffer): Buffer {
+    return input;
+}
+
+/**
+ * Validation on arguments being a big number from bn.js
+ * Throws if an argument is not in BN format or otherwise invalid
+ * @param argMap
+ */
+function validateBNLike(argMap: { [name: string]: any }) {
+    const bnLike = 'number, decimal string or BN';
+    for (const argName of Object.keys(argMap)) {
+        const argValue = argMap[argName];
+        if (argValue && !BN.isBN(argValue) && isNaN(argValue)) {
+            throw new ArgumentTypeError(argName, bnLike, argValue);
+        }
+    }
+}
+
+async function callInternal(
+    account: Account,
+    contractId: string,
+    methodName: string,
+    args: Buffer,
+    opts?: FunctionCallOptions
+): Promise<FinalExecutionOutcome> {
+    validateBNLike({ opts: opts?.gas, amount: opts?.attachedDeposit });
+
+    const rawResult = await account.functionCall({
+        contractId,
+        methodName,
+        args,
+        gas: opts?.gas,
+        attachedDeposit: opts?.attachedDeposit,
+        walletMeta: opts?.walletMeta,
+        walletCallbackUrl: opts?.walletCallbackUrl,
+        // Ignore because we have already serialized args
+        stringify: ignoreSerialization,
+    });
+
+    return getTransactionLastResult(rawResult);
+}
+
+async function viewInternal(
+    connection: Connection,
+    contractId: string,
+    functionName: string,
+    args: Buffer
+): Promise<Buffer> {
+    //* Not ideal that this logic is copied, but NAJ requires an Account to
+    //* perform a view call, which isn't necessary.
+    const serializedArgs = args.toString('base64');
+
+    const result = await connection.provider.query<CodeResult>({
+        request_type: 'call_function',
+        account_id: contractId,
+        method_name: functionName,
+        args_base64: serializedArgs,
+        finality: 'optimistic',
+    });
+
+    if (result.logs) {
+        if (!process.env['NEAR_NO_LOGS']) {
+            for (const log of result.logs) {
+                console.log(`Log [${contractId}]: ${log}`);
+            }
+        }
+    }
```

Comment:
IMO this looks a bit cleaner

```suggestion
    if (result.logs && !process.env['NEAR_NO_LOGS']) {
	  for (const log of result.logs) {
	      console.log(`Log [${contractId}]: ${log}`);
	  }
    }
```